src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -36,28 +36,44 @@ class AppTasksResource @Inject() (service: MarathonSchedulerService,
 
   @GET
   @Timed
-  def indexJson(@PathParam("appId") appId: String,
+  def indexJson(@PathParam("appId") id: String,
                 @Context req: HttpServletRequest, @Context resp: HttpServletResponse): Response = {
-    doIfAuthorized(req, resp, ViewAppOrGroup, appId.toRootPath) { implicit principal =>
+    doIfAuthenticated(req) { identity =>
       val taskMap = taskTracker.tasksByAppSync
 
-      def tasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+      def runningTasks(appIds: Set[PathId]): Set[EnrichedTask] = for {
+        runningApps <- appIds.filter(taskMap.hasAppTasks)
         id <- appIds
         health = result(healthCheckManager.statuses(id))
         task <- taskMap.appTasks(id)
       } yield EnrichedTask(id, task, health.getOrElse(task.getId, Nil))
 
-      val matchingApps = appId match {
+      id match {
         case GroupTasks(gid) =>
-          result(groupManager.group(gid.toRootPath))
-            .map(_.transitiveApps.map(_.id))
-            .getOrElse(Set.empty)
-        case _ => Set(appId.toRootPath)
+          val groupPath = gid.toRootPath
+          result(groupManager.group(groupPath)) match {
```

Comment:
I see this functionality as recurring scheme:
``` scala
option map {
  case Some(xx) => response
  case None => 404
}
```
Perhaps we can do something like:
```scala
withExisting(option, ifNot="xxx not found") { value =>  yyy}
```